README.md

```diff
@@ -771,28 +777,85 @@ let numbers = create([Number], [1, 2, 3, 4]);
+
+## Memoized Getters
+
+Microstates are immutable which makes it safe for us to memoize computations that are derived off their state. Identies will automatically memoize getters and return previously computed value when the microstate backing the identity has not changed. When a transition is invoked on the identity, the part of the identity tree that are changed will be re-created effectively invalidating the cache for the changed parts of the identity. The getters will be recomputed for state that is changed.
+
+## Debounce no-op transitions
+
+Identities automatically prevent unnecessary re-renders by debouncing transitions that do not change the value. This eliminates the need for `shouldComponentUpdate` hooks for pure components because it is safe to assume that a component that is re-rendering is re-rendering as a result of a transition that changed the value.
+
+## Identity Constructors
+
+Microstates comes with two Identity constructors: _Store_ and _Observable_. Observable will create a stream of identities. Next identity will be sent to the observer when an transition on an identity is called.
+
+### Store(microstate, callback)
+
+Store identity constructor takes two arguments: microstate and a callback. It returns an identity. When a transition is invoked on the identity, the callback will receive the next identity.
+
+```js
+import { Store, from, valueOf } from 'microstates';
+
+let initial = create(Number, 42);
+
+let last;
+
+last = Store(initial, next => (last = next));
+
+last.increment();
+//> undefined
+// callback will be invoked syncronously on transition
+
+// last here will reference the last
+last.increment();
+//> undefined
+
+valueOf(last);
+//> 44
+```
+
+The same mechanism can be used with React or any other reactive environment.
+
+```js
+import React from 'react';
+import { Store, create } from 'microstates';
+
+class Counter extends React.Component {
+  state = {
+    last: Store(create(Number, 42), next => this.setState({ last: next }))
+  };
+  render() {
+    let { last } = this.state;
+    return (
+      <button onClick={() => last.increment()}>Increment {last.state}</button>
+    );
+  }
+}
+```
+
+### Observable.from(microstate)
+
+Microstates provides an easy way to convert a Microstate which represents a single value into a Observable stream of values. This is done by passing a Microstate to Observable.from function. This function will return a Observable object with a subscribe method. You can subscribe to the stream by passing an observer to the subscribe function. Once you subscribe, you will synchronously receive a microstate with middleware installed that will cause the result of transitions to be pushed through the stream.
+
+You should be able to use to any implementation of Observables that supports Observer.from using symbol-observable. We'll use RxJS for our example.
 
 ```js
-import { from } from "rxjs";
-import { create } from "microstates";
+import { from } from 'rxjs';
```

Comment:
Not sure if you want to do this in the documentation, but in my own code that uses Microstates and RXJS I did this:

```javascript
import { from as createObservable } from 'rxjs';
```

just because `microstates` _also_ has a `from` function and I wanted to make it more clear which was being used at a glance.

Reply:
I'll do this: `import { from as observableFrom } from 'rxjs'`